numpy/core/src/common/npy_import.h

```diff
@@ -19,7 +19,7 @@
 NPY_INLINE static void
 npy_cache_import(const char *module, const char *attr, PyObject **cache)
 {
-    if (*cache == NULL) {
+    if (NPY_UNLIKELY(*cache == NULL)) {
```

Comment:
Can't hurt (unless we forgot the static somewhere ;)). Ideally, everything inside this if should probably be a non-inlined function at which point the `NPY_UNLIKELY` would not make a difference in any case (likely its not possible to measure it here already to be honest.)